include/kernel/sporadic.h

```diff
@@ -159,7 +159,7 @@ static inline bool_t sc_released(sched_context_t *sc)
  */
 static inline bool_t sc_sporadic(sched_context_t *sc)
 {
-    return sc != NULL && sc->scSporadic;
+    return sc != NULL && sc_active(sc) && sc->scSporadic;
```

Comment:
This surely works. But this would duplicate the `sc_active` test when the guard actually checks both sporadic and active, just to mention. I guess it doesn’t hurt.